pkg/apis/core/types.go

```diff
@@ -2189,37 +2189,31 @@ type Container struct {
 }
 
 // ProbeHandler defines a specific action that should be taken in a probe.
-// This type has a strong relationship to LifecycleHandler - overlapping fields
-// should be identical.
-// TODO: pass structured data to these actions, and document that data here.
+// One and only one of the following should be specified.
 type ProbeHandler struct {
-	// One and only one of the following should be specified.
 	// Exec specifies the action to take.
 	// +optional
 	Exec *ExecAction
 	// HTTPGet specifies the http request to perform.
 	// +optional
 	HTTPGet *HTTPGetAction
 	// TCPSocket specifies an action involving a TCP port.
-	// TODO: implement a realistic TCP lifecycle hook
 	// +optional
 	TCPSocket *TCPSocketAction
 }
 
 // LifecycleHandler defines a specific action that should be taken in a lifecycle
-// hook.  This type has a strong relationship to ProbeHandler - overlapping fields
-// should be identical.
-// TODO: pass structured data to these actions, and document that data here.
+// hook. One and only one of the fields, except TCPSocket should be specified.
 type LifecycleHandler struct {
-	// One and only one of the following should be specified.
 	// Exec specifies the action to take.
 	// +optional
 	Exec *ExecAction
 	// HTTPGet specifies the http request to perform.
 	// +optional
 	HTTPGet *HTTPGetAction
-	// TCPSocket specifies an action involving a TCP port.
-	// TODO: implement a realistic TCP lifecycle hook
+	// Deprecated. TCPSocket is NOT supported as a LifecycleHandler and kept
```

Comment:
followed example of some other fields. Interestingly, Deprecated with dot does not actually mark the field as deprecated. Deprecated with semicolon does, but makes generated code fail the static check.

Reply:
What error?  We should fix that.

Reply:
`./hack/verify-staticcheck.sh`:

```
Errors from staticcheck:
pkg/apis/core/v1/zz_generated.conversion.go:4127:2: out.TCPSocket is deprecated: TCPSocket is NOT supported as a LifecycleHandler and kept for the backward compatibility. There are no validation of this field and lifecycle hooks will fail in runtime when tcp handler is specified. +optional  (SA1019)
pkg/apis/core/v1/zz_generated.conversion.go:4139:55: in.TCPSocket is deprecated: TCPSocket is NOT supported as a LifecycleHandler and kept for the backward compatibility. There are no validation of this field and lifecycle hooks will fail in runtime when tcp handler is specified. +optional  (SA1019)
pkg/apis/core/validation/validation.go:2697:14: lh.TCPSocket is deprecated: TCPSocket is NOT supported as a LifecycleHandler and kept for the backward compatibility. There are no validation of this field and lifecycle hooks will fail in runtime when tcp handler is specified. +optional  (SA1019)
```

Reply:
maybe we can exclude generated files from the static check?

Reply:
yes, I think that's in the works.  Can you file a cleanup bug to come fix this after staticcheck changes get merged?

Reply:
https://github.com/kubernetes/kubernetes/issues/106405